src/main/java/com/aws/greengrass/status/FleetStatusService.java

```diff
@@ -156,46 +161,60 @@ public FleetStatusService(Topics topics, MqttClient mqttClient, DeploymentStatus
         TestFeatureParameters.registerHandlerCallback(this.getName(), this::handleTestFeatureParametersHandlerChange);
 
-        if (!deviceConfiguration.isDeviceConfiguredToTalkToCloud()) {
-            this.isConnected.set(false);
-            // Right now the connection cannot be brought online without a restart.
-            // Skip setting up scheduled upload and event triggered upload because they won't work
-            return;
-        }
-
-        Topics configurationTopics = deviceConfiguration.getStatusConfigurationTopics();
-        configurationTopics.lookup(FLEET_STATUS_PERIODIC_PUBLISH_INTERVAL_SEC)
-                .dflt(DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC)
-                .subscribe((why, newv) -> {
-                    int newPeriodicUpdateIntervalSec = Coerce.toInt(newv);
-                    // Do not update the scheduled interval if it is less than the default.
-                    if (newPeriodicUpdateIntervalSec < DEFAULT_PERIODIC_PUBLISH_INTERVAL_SEC) {
-                        return;
-                    }
-                    this.periodicPublishIntervalSec = TestFeatureParameters.retrieveWithDefault(Double.class,
-                            FLEET_STATUS_TEST_PERIODIC_UPDATE_INTERVAL_SEC, newPeriodicUpdateIntervalSec).intValue();
-                    if (periodicUpdateFuture != null) {
-                        schedulePeriodicFleetStatusDataUpdate(false);
-                    }
-                });
-
-        topics.getContext().addGlobalStateChangeListener(this::handleServiceStateChange);
-
-        this.deploymentStatusKeeper.registerDeploymentStatusConsumer(IOT_JOBS,
-                this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-        this.deploymentStatusKeeper.registerDeploymentStatusConsumer(LOCAL,
-                this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-        this.deploymentStatusKeeper.registerDeploymentStatusConsumer(SHADOW,
-                this::deploymentStatusChanged, FLEET_STATUS_SERVICE_TOPICS);
-        schedulePeriodicFleetStatusDataUpdate(false);
-
         //populating services when kernel starts up
         Instant now = Instant.now();
         this.kernel.orderedDependencies().forEach(greengrassService -> {
             serviceFssTracksMap.put(greengrassService, now);
         });
     }
 
+    @Override
+    public void postInject() {
+        super.postInject();
+
+        deviceConfiguration.onAnyChange((what, node) -> {
+            if (node != null && what.equals(WhatHappened.childChanged)
+                    && deviceConfiguration.provisionInfoNodeChanged(node, this.isFSSSetupComplete.get())) {
+                try {
+                    // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error
+                    // message to user
+                    deviceConfiguration.validate();
+
+                    Topics configurationTopics = deviceConfiguration.getStatusConfigurationTopics();
```

Comment:
Similar to `IotJobsHelper#connectToIotJobs`, we'll want to wrap these setup logic in a method and also run it once here in `postInject` even if there's callback from deviceConfiguration change. Because deviceConfig could be already setup and there's no change. Need to handle that case.

Reply:
Good point. Will wrap setup logic in a separate function and call it outside of the watcher.